### ISSUE TITLE
Memory not being released on destroying activity

### ISSUE THREAD

**Issue by divyas93** [PierfrancescoSoffritti/android-youtube-player#459]
I have observed that when activity is destroyed then all the instances of the library code are not finished and thus the memory taken by library code isn't released.

Could you please look into it.

**Comment by PierfrancescoSoffritti** [comment]
Hey, how do you know this is happening? 
Feel free to send a pull request.

**Comment by acvt007** [comment]
I use your library as an item in my RecyclerView in an Activity.
Below is my LeakCanary info:

ApplicationLeak(className=com.abc.ArticleDetailActivity, leakTrace=
┬
├─ org.chromium.device.power_save_blocker.PowerSaveBlocker
│    Leaking: NO (a class is never leaking)
│    GC Root: Global variable in native code
│    ↓ static PowerSaveBlocker.b
│                              ~
├─ java.util.HashMap
│    Leaking: UNKNOWN
│    ↓ HashMap.table
│              ~~~~~
├─ java.util.HashMap$Node[]
│    Leaking: UNKNOWN
│    ↓ array HashMap$Node[].[0]
│                           ~~~
├─ java.util.HashMap$Node
│    Leaking: UNKNOWN
│    ↓ HashMap$Node.key
│                   ~~~
├─ com.pierfrancescosoffritti.androidyoutubeplayer.core.player.views.WebViewYouTubePlayer
│    Leaking: YES (View.mContext references a destroyed activity)
│    mContext instance of com.abc.ArticleDetailActivity with mDestroyed = true
│    View#mParent is null
│    View#mAttachInfo is null (view detached)
│    View.mWindowAttachCount = 2
│    ↓ WebViewYouTubePlayer.mContext
╰→ com.theepochtimes.news.features.articledetails.ArticleDetailActivity
​     Leaking: YES (WebViewYouTubePlayer↑ is leaking and Activity#mDestroyed is true and ObjectWatcher was watching this)
​     key = 4ee2961d-1584-4a7a-8eab-d2c91687a56a
​     watchDurationMillis = 6575
​     retainedDurationMillis = 1567
, retainedHeapByteSize=295071)

**Comment by IvanLu101** [comment]
Hello, 
This problem can be reproduced in Sample App.
After adding debugImplementation 'com.squareup.leakcanary:leakcanary-android:2.0-beta-3' into build.gradle of core-sample-app, launch "ayp API" and click "Fragment Example". Then press back key to go back to main page. You will see leakcanary reports the leak(refer to the attachment).
![image](https://user-images.githubusercontent.com/11712508/68279902-6743e100-00af-11ea-88ff-7c38cf9f1c07.png)

**Comment by PierfrancescoSoffritti** [comment]
Thanks for the info. These days I don't have much time to work on the library, pull requests are welcome!

**Comment by samankit** [comment]
> I use your library as an item in my RecyclerView in an Activity. Maybe you need to check the internal webview.
> 
> Below is my LeakCanary info:
> 
> ApplicationLeak(className=com.abc.ArticleDetailActivity, leakTrace=
> ┬
> ├─ org.chromium.device.power_save_blocker.PowerSaveBlocker
> │ Leaking: NO (a class is never leaking)
> │ GC Root: Global variable in native code
> │ ↓ static PowerSaveBlocker.b
> │ ~
> ├─ java.util.HashMap
> │ Leaking: UNKNOWN
> │ ↓ HashMap.table
> │ ~~~~~
> ├─ java.util.HashMap$Node[]
> │ Leaking: UNKNOWN
> │ ↓ array HashMap$Node[].[0]
> │ ~~~
> ├─ java.util.HashMap$Node
> │ Leaking: UNKNOWN
> │ ↓ HashMap$Node.key
> │ ~~~
> ├─ com.pierfrancescosoffritti.androidyoutubeplayer.core.player.views.WebViewYouTubePlayer
> │ Leaking: YES (View.mContext references a destroyed activity)
> │ mContext instance of com.abc.ArticleDetailActivity with mDestroyed = true
> │ View#mParent is null
> │ View#mAttachInfo is null (view detached)
> │ View.mWindowAttachCount = 2
> │ ↓ WebViewYouTubePlayer.mContext
> ╰→ com.abc.ArticleDetailActivity
> ​ Leaking: YES (WebViewYouTubePlayer↑ is leaking and Activity#mDestroyed is true and ObjectWatcher was watching this)
> ​ key = 4ee2961d-1584-4a7a-8eab-d2c91687a56a
> ​ watchDurationMillis = 6575
> ​ retainedDurationMillis = 1567
> , retainedHeapByteSize=295071)

same problem with me

**Comment by samankit** [comment]
![Screenshot_2020-03-24-00-36-08-99](https://user-images.githubusercontent.com/61191969/77395717-2f240d00-6dc8-11ea-96ed-30c83b782fcb.jpg)
how to resolve it?

**Comment by aalloul** [comment]
I'm also seeing the same issue reported by Logcat on Android Studio. A workaround that seems to work is to override onStop in my fragment as follows

```
override fun onStop() {
        super.onStop()
        youtubePlayerView.release()
    }
```
I understand this is not the best option as I'm doing more than needed but at least it unregisters the `networkListener`

Proposed fix is to simply call `context.unregisterReceiver(networkListener)` in the `onStop` method of `LegacyYoutubePlayerView` ... 

@PierfrancescoSoffritti I have created a PR for a proposed fix... this is my 1st PR ever to an open-source project so I hope I'm doing things right.

Regards,
Adam

**Comment by PierfrancescoSoffritti** [comment]
This leak should be fixed in version 12.0.0 of the library.
